src/include/socket.h

```diff
@@ -312,30 +312,123 @@ static int findInterfaces(char* ifNames, union socketAddress *ifAddrs, int ifNam
   return nIfs;
 }
 
+static ncclResult_t getPortRange(unsigned short *portMin, unsigned short *portMax, bool *portRangeValid) {
+  char *portRange = getenv("NCCL_PORT_RANGE");
+
+  if (portRange == NULL) {
+    *portRangeValid = false;
+    return ncclSuccess;
+  }
+
+  int ret = sscanf(portRange, "%hu:%hu", portMin, portMax);
+  if (ret != 2 || *portMax < *portMin) {
+    WARN("Net : invalid NCCL_PORT_RANGE : %s", portRange);
+    return ncclInvalidArgument;
+  }
+
+  INFO(NCCL_ALL, "Net : Socket port range : %hu:%hu", *portMin, *portMax);
+
+  *portRangeValid = true;
+  return ncclSuccess;
+}
+
+static ncclResult_t setSocketReuse(int sockfd, bool reuseport) {
+  int opt = 1;
+
+  int optname = SO_REUSEADDR;
+
+#if defined(SO_REUSEPORT)
+  if (reuseport){
+    optname |= SO_REUSEPORT;
+  }
+#endif
+
+  SYSCHECK(setsockopt(sockfd, SOL_SOCKET, optname, &opt, sizeof(opt)), "setsockopt");
+
+  return ncclSuccess;
+}
+
+/* A simple wrapper around bind(2) which also sets the port */
+static int bindToPort(int sockfd, const union socketAddress *addr, unsigned short port) {
+  /* Create a copy of the addr with the port set */
+  union socketAddress portAddr;
+  memcpy(&portAddr, addr, sizeof(portAddr));
+  if (family == AF_INET) {
+    portAddr.sin.sin_port = htons((unsigned int)i);
```

Comment:
`htons(...port)`?

Reply:
Yup... looks like the version I compiled was not the version I pushed...